### PR TITLE
feat(pypi): update release workflow to use OIDC auth

### PR DESCRIPTION
**Pull request by esolitos**
## What

Update the "PyPi publishing" workflow to use OIDC authentication instead of personal-token.

## Why

This removes the need of storing a secret with `PYPI_TOKEN_PYVESPA` in it by using the [Trusted publishing](https://docs.pypi.org/trusted-publishers/) feature offered by PyPi.

---

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by esolitos** [comment]
Set to Draft as it requires adding the `pypa/gh-action-pypi-publish` action to the allow list.

**Comment by thomasht86** [comment]
We should also do this for [vespacli](https://github.com/vespa-engine/pyvespa/blob/master/.github/workflows/vespacli.yml)

**Comment by esolitos** [comment]
`pypa/gh-action-pypi-publish` has been approved.

I will update also vespacli workflow.

File: .github/workflows/release-pyvespa.yml [hunk 1]
```diff
@@ -7,9 +7,21 @@ on:
 jobs:
   update_version:
     runs-on: ubuntu-latest
+
+    # The environment is reuiqred for OIDC to work.
```
Comment (thomasht86) [inline .github/workflows/release-pyvespa.yml:11]:
```suggestion
    # The environment is required for OIDC to work.
```

File: .github/workflows/release-vespacli.yml [hunk 2]
```diff
@@ -104,45 +124,25 @@ jobs:
...
-      - name: Set up Python
-        uses: actions/setup-python@v5
-        with:
-          python-version: "3.10"
-     
-      - name: Install dependencies and this repo
-        run: |
-          python -m pip install --upgrade pip
-          pip install -e .[build]
+    # The environment is reuiqred for OIDC to work.
```
Comment (thomasht86) [inline .github/workflows/release-vespacli.yml:130]:
```suggestion
    # The environment is required for OIDC to work.
```

File: vespacli/utils/check_latest_version.py [hunk 3]
```diff
@@ -1,13 +1,28 @@
...
 import sys
+import requests
+from packaging import version
 
-if __name__ == "__main__":
+
+def get_latest_pypi_version() -> version.Version:
+    response = requests.get(
+        "https://pypi.org/simple/vespacli/",
+        headers={"Accept": "application/vnd.pypi.simple.v1+json"}
```
Comment (thomasht86) [inline vespacli/utils/check_latest_version.py:10]: This was nice and clean!

**Review by thomasht86 — DISMISSED** [review]
Very nice!

**Review by thomasht86 — APPROVED** [review]
👍

**Pull request merged**